### DISCUSSION TITLE
Mounting a sub directory of my solution into the docker container

### DISCUSSION THREAD

**Discussion by dgiddins** [testcontainers/testcontainers-dotnet#767] [Q&A] [answered]
Hi,

We use sqitch for managing schema deployments into our postgres database. I want to have a testfixture that first runs postgres in test containers then spin up another testcontainer that contains an installation of sqitch and pass it our sqitch files so that we can deploy our schema for testing. 

I've got my postgres instance running but right now I am struggling to make any head way with running my sqitch files in testcontainers. 

So far I have the following:

```csharp
_sqitchContainer = new ContainerBuilder()
                .WithName(Guid.NewGuid().ToString("D"))
                .WithImage("docker.otenv.com/location-docker-sqitch")
                .WithVolumeMount("//c/github.com/location-scheduled-tasks/location-read-schema", "/mnt/location-read-schema")
                .WithWorkingDirectory("/mnt/location-read-schema")
                .WithWaitStrategy(Wait.ForUnixContainer().UntilCommandIsCompleted("deploy"))
                .Build();
```

I get the following.

    Docker.DotNet.DockerApiException : Docker API responded with status code=BadRequest, response={"message":"create //c/github.com/opentable/location-scheduled-tasks/location-read-schema: \"//c/github.com/location-scheduled-tasks/location-read-schema\" includes invalid characters for a local volume name, only \"[a-zA-Z0-9][a-zA-Z0-9_.-]\" are allowed. If you intended to pass a host directory, use absolute path"}

if i change to 

```csharp
_sqitchContainer = new ContainerBuilder()
                .WithName(Guid.NewGuid().ToString("D"))
                .WithImage("docker.otenv.com/location-docker-sqitch")
                .WithVolumeMount("//c/github.com/location-scheduled-tasks/location-read-schema", "location-read-schema")
                .WithWorkingDirectory("location-read-schema")
                .WithWaitStrategy(Wait.ForUnixContainer().UntilCommandIsCompleted("deploy"))
                .Build();
```

And I get the following

    Docker.DotNet.DockerApiException : Docker API responded with status code=BadRequest, response={"message":"the working directory 'location-read-schema' is invalid, it needs to be an absolute path"}

The docker command  

```
docker run -it --volume //c/github.com/location-scheduled-tasks/location-read-schema://location-read-schema -w //location-read-schema location-docker-sqitch
```

Works. 

I'm confused by the API for testcontainers vs how it seems to be working in Docker command line.

**Comment by HofmeisterAn** [comment, marked as answer]
`WithVolumeMount(string, string)` is intended for Docker managed volumes. You are lookin for `WithBindMount(string, string)`. Please note, that we do not recommend to mount local paths into the container. It makes testing very hard across different development machines or CI environments.

**Reply by dgiddins** [reply]
Thank you.

**Reply by HofmeisterAn** [reply]
Out of completes, you might wanna look at [WithResourceMapping(string, string)](https://github.com/testcontainers/testcontainers-dotnet/blob/7ad541469cb71f4cfb472bf500858bddae7cc479/tests/Testcontainers.Tests/Unit/Containers/Unix/CopyResourceMappingContainerTest.cs#L28-L29) or one of its overloaded members. It copies a file into the created container. Depending on how you organize your schema migration you can get rid of mounting the local path.

**Reply by dgiddins** [reply]
I may well look at that as an option, the only difficulty will be in the number of files that need to be moved so I would need to scan programatically. I'll see how well this work in CI first.

Thank you again - I wasn't expecting such a fast response.
